cmd/transportRequestUploadRFC_test.go

```diff
@@ -50,61 +52,90 @@ func (m *uploadMock) WithConnection(c rfc.Connection) {
 
 func (m *uploadMock) Perform(exec rfc.Exec) error {
 	m.uploadCalled = true
-	return nil
+	return m.failWith
 }
```

Comment:
So that I understand better- Perform is defined here as a mock version of the function- is that because the testing function can't access the regular function? Where is the regular function? I noticed that it's also called from the regular code (non test) on line 91 of the new version of transportRequestUploadRFC.go.

Reply:
The regular function is located in pkg/transportrequest/rfc/upload.go.
runTransportRequestUploadRFC() is the mockable layer. the unit tests call this method and mocks the parameters while the command implementation transportRequestUploadRFC passes the regular objects.
All GO steps/commands are implemented like this. There is a generator which creates the main functions.

Reply:
Thank you!